src/prefect/deployments.py

```diff
@@ -40,16 +32,29 @@ async def load_flow_from_flow_run(
     Load a flow from the location/script provided in a deployment's storage document.
     """
     deployment = await client.read_deployment(flow_run.deployment_id)
-    storage_document = await client.read_block_document(deployment.storage_document_id)
-    storage_block = Block._from_block_document(storage_document)
+    if deployment.storage_document_id:
+        storage_document = await client.read_block_document(
+            deployment.storage_document_id
+        )
+        storage_block = Block._from_block_document(storage_document)
+    else:
+        basepath = deployment.path or Path(deployment.manifest_path).parent
+        storage_block = LocalFileSystem(basepath=basepath)
 
     sys.path.insert(0, ".")
+    # TODO: append deployment.path
```

Comment:
👀

Reply:
I'm trying to keep the PRs as small as possible (so this PR includes the data change and removal of `manifest.json` mainly), and the additional `path` manipulation will come in a separate PR